lib/facter/java_path.rb

```diff
@@ -0,0 +1,19 @@
+# Fact: java_path
+#
+# Purpose: get absolute path of java executable
+#
+# Resolution:
+#   Uses `readlink` to resolve the path of `/usr/bin/java`
+#
+# Caveats:
+#   Requires readlink
+#
+# Notes:
+#   None
+Facter.add(:java_path) do
+  setcode do
+    if Facter::Util::Resolution.which('readlink')
+      Facter::Util::Resolution.exec('readlink -f /usr/bin/java').strip
```

Comment:
Instead of calling out to `readlink`, which may or may not exist on a system, use [`File#readlink`](http://ruby-doc.org/core-1.8.7/File.html#method-c-readlink), which is built into Ruby.